cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -853,15 +858,44 @@ Future<ReconciliationState> reconcileEntityOperator(Clock clock)    {
         }
 
         /**
-         * @return true if the autorebalance is enabled within the Cruise Control configuration, false otherwise
+         * Based on the autorebalance being enabled or not within Cruise Control configuration:
+         * - return true because the auto-rebalancing reconciliation has to run or ...
+         * - ... create the auto-rebalance status on Idle if the cluster is still in the creation phase (no auto-rebalancing to run yet) or ...
+         * - ... reset to null the auto-rebalance status because the auto-rebalance was disabled
+         *
+         * @return true if the autorebalance reconciliation has to run, false otherwise
          */
-        boolean isAutoRebalancingEnabled() {
-            // Cruise Control is not defined in the Kafka custom resource or no autorebalance enabled, so nothing to reconcile
+        boolean maybeAutoRebalancing() {
             if (kafkaAssembly.getSpec().getCruiseControl() == null || kafkaAssembly.getSpec().getCruiseControl().getAutoRebalance() == null) {
                 LOGGER.debugCr(reconciliation, "Cruise Control or inner autorebalance field not defined in the Kafka custom resource, no auto-rebalancing to reconcile");
+                // enforce no auto-rebalance status, if we are disabling Cruise Control and/or auto-rebalance from its configuration
+                kafkaStatus.setAutoRebalance(null);
                 return false;
+            } else {
+                // there is an auto-rebalance status already, so we need to run an auto-rebalance reconciliation
+                if (kafkaAssembly.getStatus() != null && kafkaAssembly.getStatus().getAutoRebalance() != null) {
```

Comment:
This seems wrong? What exactly does this condition try to detect? What if I enable auto-rebalancing and do a scale-up in one step?

Reply:
> This seems wrong? What exactly does this condition try to detect?

So right now I am detecting that it's cluster creation so the auto-rebalance reconciliation doesn't need to run thanks to the absence of the `status.autoRebalance` which has to be initialized to `Idle` (as it happens in the "else"). When the `status.autoRebalance` is there, it means that it's not cluster creation and auto-rebalancing reconcile has to run.

> What if I enable auto-rebalancing and do a scale-up in one step?

That's a good point I didn't consider until now. It actually brakes the above rule. No `status.autoRebalance` but we need to run auto-rebalancing reconcile anyway because it's not cluster creation. I will think about it ...

Reply:
This should be now fixed.